src/kubernetes_configmap_jwt_scan/kubernetes_configmap_jwt_scan/scan.py

```diff
@@ -0,0 +1,56 @@
+import datetime
+import logging
+import os
+import re
+import shutil
+import tarfile
+import jwt
+
+from distutils.dir_util import mkpath
+
+import kubernetes_configmap_jwt_scan.context
+
+from kubernetes_configmap_jwt_scan.config import config
+from lib.kctl import Kctl
+
+WARN_THRESHOLD = 30 # warn if expiring within 30 days
+JWT_REGEX = re.compile('^(?:[\w-]*\.){2}[\w-]*$')
+
+def is_jwt(str):
+  if not JWT_REGEX.match(str):
+    return False
+  header = None
+  try:
+    header = jwt.get_unverified_header(str)
+  except jwt.DecodeError as e:
+    return False
+  return header is not None
+
+def scan():
+  logging.info(f"Scanning {config.artsy_env}...")
+  kctl = Kctl(config.in_cluster, config.artsy_env)
+  results = {}
+  now = datetime.datetime.now()
+  for configmap in kctl.get_configmaps():
+    if 'data' not in configmap:
+      continue
+    for key, val in configmap['data'].items():
+      if is_jwt(val):
+        logging.debug(f"Found {key} in {configmap['metadata']['name']} that looks like a JWT")
+        payload = jwt.decode(val, options={"verify_signature": False, "verify_exp": False})
+        if 'exp' in payload:
+          exp_date = datetime.datetime.utcfromtimestamp(payload['exp'])
+          days_to_expiry = (exp_date - now).days
+          if days_to_expiry <= WARN_THRESHOLD:
+            details = {
+              'days_left': days_to_expiry,
+              'aud': payload.get('aud'),
+              'subject_application': payload.get('subject_application')
+            }
+            logging.error(f"Found expiring {key} in {configmap['metadata']['name']}: {details}")
+            if configmap['metadata']['name'] not in results:
+              results[configmap['metadata']['name']] = {}
+            results[configmap['metadata']['name']][key] = details
+  if results:
+    raise SystemExit('Some tokens are expiring!')
```

Comment:
Oh nice, didn't know about SystemExit. About exiting the program here (or anywhere not in the topmost script), I as well did that for the other scripts, but decided it's safer to just raise an exception, in case there's code upstream that would like to catch the exception and do some remediation.

Reply:
Oh, I meant to link to an [example](https://github.com/artsy/opstools/blob/b05d7c07612d40bcaea9555e27db5c29fea82f94/src/kubernetes_export/kubernetes_export/config.py#L67).